pkg/cmd/datastore/datastore.go

```diff
@@ -111,11 +152,13 @@ func RegisterDatastoreFlagsWithPrefix(flagSet *pflag.FlagSet, prefix string, opt
 
 	flagSet.StringVar(&opts.Engine, flagName("datastore-engine"), defaults.Engine, fmt.Sprintf(`type of datastore to initialize (%s)`, datastore.EngineOptions()))
 	flagSet.StringVar(&opts.URI, flagName("datastore-conn-uri"), defaults.URI, `connection string used by remote datastores (e.g. "postgres://postgres:password@localhost:5432/spicedb")`)
-	flagSet.IntVar(&opts.MaxOpenConns, flagName("datastore-conn-max-open"), defaults.MaxOpenConns, "number of concurrent connections open in a remote datastore's connection pool")
-	flagSet.IntVar(&opts.MinOpenConns, flagName("datastore-conn-min-open"), defaults.MinOpenConns, "number of minimum concurrent connections open in a remote datastore's connection pool")
-	flagSet.DurationVar(&opts.MaxLifetime, flagName("datastore-conn-max-lifetime"), defaults.MaxLifetime, "maximum amount of time a connection can live in a remote datastore's connection pool")
-	flagSet.DurationVar(&opts.MaxIdleTime, flagName("datastore-conn-max-idletime"), defaults.MaxIdleTime, "maximum amount of time a connection can idle in a remote datastore's connection pool")
-	flagSet.DurationVar(&opts.HealthCheckPeriod, flagName("datastore-conn-healthcheck-interval"), defaults.HealthCheckPeriod, "time between a remote datastore's connection pool health checks")
+
+	var legacyConnPool ConnPoolConfig
+	RegisterConnPoolFlagsWithPrefix(flagSet, "datastore-conn", DefaultConnPoolConfig(), &legacyConnPool)
+	deprecateUnifiedConnFlags(flagSet)
+	RegisterConnPoolFlagsWithPrefix(flagSet, "datastore-connpool-read", &legacyConnPool, &opts.ReadConnPool)
```

Comment:
why are we using `legacyConnPool` here as the default?

Reply:
Basically to support mixing and matching flags. The new flags prioritize over the old ones, so if you have `--datastore-conn-healthcheck-interval` and `--datastore-connpool-read-healthcheck-interval`, the latter takes precedent over the former.